source/common/upstream/health_checker_impl.cc

```diff
@@ -217,8 +217,11 @@ TcpHealthCheckMatcher::MatchSegments TcpHealthCheckMatcher::loadProtoBytes(
   MatchSegments result;
 
   for (const auto& entry : byte_array) {
-    const std::string& hex_string = entry.text();
-    result.push_back(Hex::decode(hex_string));
+    const auto& decoded = Hex::decode(entry.text());
```

Comment:
technically storing reference to temporary here. I would remove `&`

Reply:
Done. Interesting that the compiler didn't complain about it.

Reply:
I've seen this several times and I don't understand how the compiler doesn't complain about it either. I think it probably doesn't complain because it generates correct code. Meaning, it can infer that the temporary sticks around long enough, but IMO it should complain and error.